plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/thrift/ThriftHiveMetastore.java

```diff
@@ -1907,12 +1919,14 @@ public void updateTableWriteId(HiveIdentity identity, String dbName, String tabl
         requireNonNull(tableName, "tableName is null");
         checkArgument(writeId > 0, "writeId should be a positive integer, but was %s", writeId);
         try {
+            Table table = getTable(identity, dbName, tableName)
+                    .orElseThrow(() -> new TableNotFoundException(new SchemaTableName(dbName, tableName)));
+            rowCountChange.ifPresent(rowCount ->
+                    table.setParameters(adjustRowCount(table.getParameters(), tableName, rowCount)));
             retry()
                     .stopOnIllegalExceptions()
                     .run("updateTableWriteId", stats.getUpdateTableWriteId().wrap(() -> {
-                        try (ThriftMetastoreClient metastoreClient = createMetastoreClient(identity)) {
-                            metastoreClient.updateTableWriteId(dbName, tableName, transactionId, writeId, rowCountChange);
-                        }
+                        alterTransactionalTable(identity, table, transactionId, writeId);
```

Comment:
I think the `Table table = getTable(identity, dbName, tableName)` should be redone under retry body

Reply:
(i am not totallt sure though)

Reply:
@findepi why do you think this should be redone under the retry body? Is there some test I could do to trigger some behavior you are worried about?

Reply:
i guess this may be fine as is. i wasn't sure (and still not fully am)